### DISCUSSION TITLE
multi-collide Bounce with fixed resting position

### DISCUSSION THREAD

**Discussion by YTN0** [bepu/bepuphysics2#275] [Q&A]
Hi,

Am new to Bepu, and I apologize in advance if this is a simple or basic question. 

Here's my scenario, I have an object in a static position (x,y,z), and I have multiple other objects hitting it within a short amount of time. I would like to bounce the hit object out by a specific distance within a specific trajectory and have it settle back to its original position (important). If it is hit by only one object, then it will bounce once and settle. However, it may be hit by N objects within a certain timeframe, and should continue bouncing until the end of the last hit.

The closest similar example I can think of is a pinball machine bumper (the circular ones in the middle of the table) that has a large visible bounce and is hit multiple times.

I am assuming I would need to use some kind of spring for this. Any pointers would be appreciated!

**Comment by RossNordby** [comment]
While you could use a dynamic body controlled by a `OneBodyLinearServo` with spring settings configured to the desired behavior, my guess is you're looking for more controllable behavior than that would offer.

You could also use a kinematic body and control its velocity arbitrarily. So, upon being touched, execute whatever behavior you want by modifying the velocity to accomplish it. To detect collisions, you could use something like the [SolverContactEnumerationDemo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/SolverContactEnumerationDemo.cs), [CollisionTrackingDemo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/CollisionTrackingDemo.cs), or [ContactEventsDemo](https://github.com/bepu/bepuphysics2/blob/master/Demos/Demos/ContactEventsDemo.cs).

**Comment by YTN0** [comment]
Thanks. WIll give that a try.
